src/precice/SolverInterface.hpp

```diff
@@ -628,6 +641,60 @@ class SolverInterface {
       const int *   valueIndices,
       const double *values);
 
+  /**
+   * @brief Writes vector gradient data given as block.
+   *
+   * This function writes values of specified vertices to a dataID.
+   * Values are provided as a block of continuous memory.
+   * valueIndices contains the indices of the vertices
+   *
+   * Per default, the values are passed as following:
+   *
+   * The 2D-format of value is (v0x_dx, v0x_dy, v0y_dx, v0y_dy,
+   *                            v1x_dx, v1x_dy, v1y_dx, v1y_dy,
+   *                            ... ,
+   *                            vnx_dx, vnx_dy, vny_dx, vny_dy)
+   *
+   * corresponding to the vector data v0 = (v0x, v0y) , v1 = (v1x, v1y), ... , vn = (vnx, vny) derived in spatial directions.
+   *
+   *
+   * The 3D-format of value is (v0x_dx, v0x_dy, v0x_dz, v0y_dx, v0y_dy, v0y_dz, v0z_dx, v0z_dy, v0z_dz,
+   *                            v1x_dx, v1x_dy, v1x_dz, v1y_dx, v1y_dy, v1y_dz, v1z_dx, v1z_dy, v1z_dz,
+   *                            ... ,
+   *                            vnx_dx, vnx_dy, vnx_dz, vny_dx, vny_dy, vny_dz, vnz_dx, vnz_dy, vnz_dz)
+   *
+   * corresponding to the vector data v0 = (v0x, v0y, v0z) , v1 = (v1x, v1y, v1z), ... , vn = (vnx, vny, vnz) derived in spatial directions.
+   *
+   * The optional rowMajor attribute allows to enter the values derived in the spatial directions first:
+   *
+   * For the 3D-format as follows: (v0x_dx, v0y_dx, v1x_dx, v1y_dx, ... , vnx_dx, vny_dx,
+   *                                v0x_dy, v0y_dy, v1x_dy, v1y_dy, ... , vnx_dy, vny_dy)
+   *
+   *
+   * For the 3D-format as follows: (v0x_dx, v0y_dx, v0z_dx, v1x_dx, v1y_dx, v1z_dx, ... , vnx_dx, vny_dx, vnz_dx,
+   *                                v0x_dy, v0y_dy, v0z_dy, v1x_dy, v1y_dy, v1z_dy, ... , vnx_dy, vny_dy, vnz_dy,
+   *                                v0x_dz, v0y_dz, v0z_dz, v1x_dz, v1y_dz, v1z_dz, ... , vnx_dz, vny_dz, vnz_dz)
+   *
+   *
+   * @param[in] dataID ID to write to.
+   * @param[in] size Number n of vertices.
+   * @param[in] values Pointer to the gradient values read columnwise by default.
+   * @param[in] rowMajor Allows to input the data derived in spatial directions first
+   *
+   * @pre count of available elements at gradient values matches the configured dimension * size
+   * @pre count of available elements at valueIndices matches the given size
```

Comment:
Since in the implementation we don't (or rather we can't) check if the pointers have the needed size, we assume that it is a pre conditition right ?